### ISSUE TITLE
SMS Text to Speech notification service

### ISSUE THREAD

**Issue by feld** [louislam/uptime-kuma#3295]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Notification

### 🔖 Feature description

The one thing missing from self hosting an alerting system is a way to get a phone to ring with a message. That's what stops me from replacing the monitoring system we use at work.

### ✔️ Solution

Maybe something like TextMagic https://www.textmagic.com

I'm still searching for other services that can fill this role.

I wondered about https://jmp.chat which is XMPP but there isn't a way to make an audio file attachment become a phone call as far as I am aware.

**Comment by CommanderStorm** [comment]
>I'm still searching for other services that can fill this role.

Twillo also has this capability

**Comment by CommanderStorm** [comment]
https://github.com/MarshalX/tgcalls was raised in #816 as an implementation for this in python for calling to telegram

**Comment by Aj7Ay** [comment]
Is this implemented so we can use Twilio for call Notification

**Comment by CommanderStorm** [comment]
No, this is not currently implemented.
We would have closed the issue if it had.

**Comment by Aj7Ay** [comment]
We can't get have voice call feature?
From Twilio
If possible
Can we know expected time for this
We will wait for it sir

**Comment by alanescarcha** [comment]
bump

**Comment by chreniuc** [comment]
I'm also interested in this, it doesn't need to be have the infrastructure for initiating those calls, it should only send to existing thirdparty apis, like specified above, eg twillo.

I want to use calls alerts for those really important things

**Comment by markwitt1** [comment]
Any update here? should be quize easy to add for native Twilio integration

**Comment by CommanderStorm** [comment]
There has not been progress on this issue to my knowlege, but you can be the change that you want to see 😉

The current twillo code is located here: https://github.com/louislam/uptime-kuma/blob/master/server/notification-providers/twilio.js

> To prevent misscomunication: we are open to such notifiation providers, here is our contribution guide: https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md

**Comment by noobnooc** [comment]
For those still wanting phone call alerts from Uptime Kuma — you can set up a webhook
notification in Uptime Kuma pointing to Echobell (https://echobell.one), which converts
webhook calls into actual iOS phone call alerts. Not native, but gets the job done.

**Comment by CommanderStorm** [comment]
@noobnooc advertising your service like this without telling that this is advertisement is very disingenuous!

**Comment by noobnooc** [comment]
> [@noobnooc](https://github.com/noobnooc) advertising your service like this without telling that this is advertisement is very disingenuous!

@CommanderStorm You're absolutely right, and I apologize for that. I should have disclosed upfront that I'm the services developer. My intention was to share a potentially helpful solution to this long-standing request, but I understand that posting without transparency about my affiliation is disingenuous. I'll make sure to be upfront about that in the future.

**Comment by oneitpro** [comment]
# ✅ Workaround: Twilio (Notification type platforms) SMS with A2P 10DLC using Webhook instead of native provider

If you're getting error **21408** or **21606** with Uptime Kuma's built-in Twilio notification provider, here's an alternative solution.


## Workaround — Use Webhook instead

1. Add a new notification → select **Webhook**
2. Fill in the fields:

**URL**
```
https://YOUR_ACCOUNT_SID:YOUR_AUTH_TOKEN@api.twilio.com/2010-04-01/Accounts/YOUR_ACCOUNT_SID/Messages.json
```

**HTTP Method:** `POST`

**Request Body:** `Custom Body`

**Body:**
```
From=%2B1XXXXXXXXXX&To=%2B1XXXXXXXXXX&Body={{ msg }}
```

**Additional Headers:**
```json
{"Content-Type": "application/x-www-form-urlencoded"}
```

3. Click **Test** — you should receive the SMS instantly.

## ⚠️ Important
- The `+` in phone numbers **must** be URL-encoded as `%2B` — without this Twilio receives a space and rejects the number with error 21606.
- Use `From=` for a specific sender number, or replace it with `MessagingServiceSid=MGXXXXXXXX` to let Twilio auto-select.
- Tested and working with A2P 10DLC approved campaigns on Twilio.

Hope this saves someone a few hours! 🙏
